### DISCUSSION TITLE
feat: add pure css back to top button

### DISCUSSION THREAD

**Discussion by parademovie** [hugo-sid/hugo-blog-awesome#29] [Feature requests]
What I like about this theme is that there is almost no js,that's great.
https://moderncss.dev/pure-css-smooth-scroll-back-to-top/
https://codepen.io/rolandtoth/pen/eMamVK
https://codepen.io/t_afif/pen/QWqdYYB

**Comment by hugo-sid** [comment]
Hi @parademovie, it's great to hear that you liked this theme having minimal JS!

I wanted to let you know that I tried implementing the back-to-top button using the CodePen links you provided, but I ran into some conflicts with the existing styles. I didn't have a chance to troubleshoot the issue thoroughly, but I thought I would let you know in case you want to take a look at it yourself.

If you have any suggestions or would like to submit a pull request, please feel free to do so.

**Comment by parademovie** [comment]
Hi @hugo-sid,
Sorry,I just glanced at the article and thought it would work for all layouts, actually it seems like it has to require `body {display: grid;}`,so if you don't want to change the whole layout,you only have to take other way.

The good news is that the alternative requires very little js.
(Note: this solution does not consider mobile adaptation and dark mode, as well as other more details)

\themes\hugo-blog-awesome\assets\sass\_base.scss
```
html{
  scroll-behavior:smooth;
}

#totop {
  display: none;
  position: fixed;
  bottom: 60px;
  right: 30px;
  z-index: 99;
  width: 0; 
  height: 0; 
  border-left: 10px solid transparent;
  border-right: 10px solid transparent;  
  border-bottom: 10px solid black;

}
```

\themes\hugo-blog-awesome\layouts\partials\footer.html
```
<a href="#" id="totop"></a>
```
\themes\hugo-blog-awesome\layouts\partials\scriptsBodyEnd.html
```
{{ $script_main := resources.Get "js/totop.js" | minify | fingerprint }}
<script src="{{ $script_main.Permalink }}"></script>
```
creat `totop.js`
\themes\hugo-blog-awesome\assets\js\totop.js
```
const mybutton = document.getElementById("totop");

window.onscroll = () => 
{
  if (document.body.scrollTop > 200 || document.documentElement.scrollTop > 200) {
    mybutton.style.display = "block";
  } else {
    mybutton.style.display = "none";
  }
}
```
Works perfectly on my pc!
Hope you have a good day!

**Reply by hugo-sid** [reply]
> it seems like it has to require `body {display: grid;}`

I faced the same problem as well.

> The good news is that the alternative requires very little js.

I agree.

**Reply by hugo-sid** [reply]
> Note: this solution does not consider mobile adaptation and dark mode, as well as other more details

I will take care of all that.

**Comment by hugo-sid** [comment]
Hi @parademovie. 

Based on you inputs, I pushed a [commit](https://github.com/hugo-sid/hugo-blog-awesome/commit/170da4f84038a764e90fc5e006dab6018a5adbe2) which implements a go to top button. The corresponding documentation is [here](https://github.com/hugo-sid/hugo-blog-awesome#enable-go-to-top-button).

Thanks for your hard work and your willingness to contribute to this theme.
Your contribution is appreciated.

Feel free to close this discussion if you don't have anything else to say regarding this feature.

Don't forget to star the repo, if you liked this theme.

Have a good day. Happy blogging :)

**Comment by parademovie** [comment]
Hi @hugo-sid,by the way, if I want to remove the code highlight from the hugo-blog-awesome theme.Should I delete `\assets\code-highlight.css` ? What other code snippets should I remove, thanks?

**Comment by hugo-sid** [comment]
@parademovie, there are several ways to write code snippets, without the code highlight:
- put 3 backticks and don't mention any language
- indent you code snippet with four spaces

If you are sure that you don't need this feature, you may:
- delete `\assets\code-highlight.css`
- remove it's reference from `head.html` [partial](https://github.com/hugo-sid/hugo-blog-awesome/blob/main/layouts/partials/head.html#L17-L18)

**Reply by parademovie** [reply]
OK, I got it.Thanks again and good luck!
